tests/fetch/test_giturl.py

```diff
@@ -36,6 +37,22 @@ def setUp(self):
             "garbage",
         )
         self.fh = giturl.GitUrlFetchHandler()
+        self._git_version_2_28_plus = None
```

Comment:
sigh -- this is why `@cached_property` should have been a thing long before `python3.8`

Reply:
Indeed.  C-h will bump major version nums as each py3.x goes EOL so that eventually we can use the newer features.